impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -329,8 +392,55 @@ public ClientBuilder setOrgUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setAuthorizationMode(String authorizationMode) {
+        // if 'authorizationMode' is ever supplied, it should only be "PrivateKey" (OAUTH2 flow);
+        // else do not set it as it defaults to SSWS flow.
+        if (authorizationMode != null) {
+            if (authorizationMode.equals("PrivateKey")) {
+                this.clientConfig.setAuthorizationMode(authorizationMode);
+            }
+            else {
+                throw new IllegalArgumentException("Invalid authorizationMode");
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setScopes(List<String> scopes) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(scopes, "Missing scopes");
+            Assert.isTrue(scopes.size() > 0, "Empty scopes list");
```

Comment:
Collapse this into a single check, so the use will see the same detailed error if the value is null or an empty string
Maybe something like at least one scope is required.

nit: we should probably add a help URL similar to the `{yourOktaDomain}` placeholder, i.e. https://bit.ly/finding-okta-domain
@bretterer @laura-rodriguez ^ probably outside the scope of this PR, but this is likely going to be a common issue? (assumption)

Reply:
FWIW I used https://bit.ly/finding-okta-app-credentials for `clientId`.